### PR TITLE
[RHCLOUD-17832] Internal authentication endpoint

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
Creates a new internal "Authentications" endpoint which exposes the password of the authentication.

* It also removes the `password` field from the public `AuthenticationResponse` objects.

## Links

[RHCLOUD-17832](https://issues.redhat.com/browse/RHCLOUD-17832)

### CODE REVIEW

**Review by lindgrenj6 — APPROVED** [review]

File: internal_handlers.go [hunk 1]
```diff
@@ -8,6 +8,21 @@ import (
...
 )
 
+// InternalAuthenticationGet fetches one authentication and returns it with the password exposed. Internal use only.
+func InternalAuthenticationGet(c echo.Context) error {
+	authDao, err := getAuthenticationDao(c)
+	if err != nil {
+		return err
+	}
+
+	auth, err := authDao.GetById(c.Param("uuid"))
```
Comment (lindgrenj6) [inline internal_handlers.go:18]: Sorry I already approved but forgot to comment, we need this functionality hidden behind that query parameter `expose_encrypted_attributes[]=password` basically do something similar to the way i did in the filtering middleware.

Heres an example request I make from the superkey worker: https://github.com/RedHatInsights/sources-superkey-worker/blob/d076299a7d68563aa563256c37c8d846412f44d3/sources/api_client.go#L58-L63

Comment (lindgrenj6) [reply internal_handlers.go:18]: Hm. actually maybe since it wasn't documented we can change it. Let me chew on it a bit more.

Comment (lpichler) [reply internal_handlers.go:18]: Does this require any other change ?

**Comment by MikelAlejoBR** [comment]
I added a small patch to it so the rewrite doesn't break the current implementations that are passing the query parameter. Nonetheless, what's the purpose of the query parameter?

**Comment by lindgrenj6** [comment]
> I added a small patch to it so the rewrite doesn't break the current implementations that are passing the query parameter. Nonetheless, what's the purpose of the query parameter?

Basically if we had multiple encrypted fields we could expose them selectively. For us there is only one so it makes less sense. Basically just an extra layer of security by obscurity for us afaik.

**Review by lindgrenj6 — APPROVED** [review]

**Comment by MikelAlejoBR** [comment]
Thank you for the explanation! :smiley:

**Pull request merged**